test/if.cc

```diff
@@ -117,8 +119,8 @@ TEST(IfTest, Raise) {
     return Just(1)
         | Then([](int i) {
              return If(i == 1)
-                 .yes(Just(i))
-                 .no(Raise("raise"));
+                 .yes([&]() { return Just(i); })
```

Comment:
What if we send `i` as an argument to this function, so we can `move` some args to eventuals